### ISSUE TITLE
Suggestion: Remove extension enforcement in file chooser

### ISSUE THREAD

**Issue by xsmile** [lah7/gtk3-classic#2]
I don't know if the removal of this feature is possible or how easily this can be realized but if it is, here is my suggestion.

In some cases, the GTK3 file chooser dialogue enforces a pre-set file extension when saving a file, even if the user wants to save a file without it and removes it from the file name. E.g. Firefox can save an extensionless text file but Chromium cannot.

Take this URL as an example and try to save the page as `PKGBUILD` without the `.txt` extension:
`https://aur.archlinux.org/cgit/aur.git/plain/PKGBUILD?h=gtk3-mushrooms`.

**Comment by TomaszGasior** [comment]
Thanks for your activity.

I made small tests with Chrome and Firefox. I can confirm problem. However I checked that when you remove manually extension from file name in GTK's file chooser (opened by Chromium) file path provided to application by GTK's file chooser doesn't have it. In other words, extension is included by Chromium itself after closing GTK's file chooser.

I am not expert in GTK and C language, so I can be wrong. Have you any other examples of apps that enforces extension in file chooser dialog?

**Comment by xsmile** [comment]
Until now I saw this behavior with Chromium only. Having tested some more GTK3 applications, Midori and Gedit do not force a file extension, so you are probably right about Chromium being the origin of it.

I tried changing Chromium's source to test the behavior of the file chooser dialogue but couldn't find any code responsible for the forced extension. Here is a link to the source, if you want to take a look:
[select_file_dialog_impl_gtk.cc#387](https://chromium.googlesource.com/chromium/src/+/61.0.3163.90/chrome/browser/ui/libgtkui/select_file_dialog_impl_gtk.cc#387). My GTK expertise is missing completely.

Thanks for the quick reply.
